src/main/java/com/redhat/cloud/common/clowder/configsource/ClowderConfigSourceFactory.java

```diff
@@ -37,13 +39,25 @@
         // supply them to our source.
         Map<String, ConfigValue> exProp = new HashMap<>();
         Iterator<String> stringIterator = configSourceContext.iterateNames();
-        while (stringIterator.hasNext()) {
-            String key = stringIterator.next();
-            ConfigValue value = configSourceContext.getValue(key);
-            exProp.put(key,value);
-        }
 
-        return Collections.singletonList(new com.redhat.cloud.common.clowder.configsource.ClowderConfigSource(clowderConfig, exProp));
+        /*
+         * A config value expansion happens when an expression wrapped into `${ }` is resolved.
+         * For example, if the configuration contains `foo=1234` and `bar=${foo}`, then `bar`
+         * will be resolved and expanded to the value `1234`.
+         * However, if an expression cannot be resolved because it refers to a missing config key,
+         * then the expansion fails and a NoSuchElementException is thrown.
+         * Such a throw should only happen when the config value is actually used and not here.
+         * That's why we need to disable the config values expansion.
+         */
+        withoutExpansion(() -> {
+            while (stringIterator.hasNext()) {
+                String key = stringIterator.next();
+                ConfigValue value = configSourceContext.getValue(key);
```

Comment:
Wouldn't the `withoutExpansion` affect expressions in values?

Reply:
I'm not sure I understand that question.

`withoutExpansion` prevents the expressions expansion in this piece of code but the expressions will still be resolved when the config values are used by your application.

Reply:
Values here are being collected and passed to `ClowderConfigSource` as `existingValues`, wouldn't the `withoutExpansion` have an impact for such existing value that should have been resolved. Within `ClowderConfigSource` it would be getting raw, non-resolved, value, wouldn't it? Sorry, I have limited knowledge on how or why the `existingValues` are being used.

Reply:
> Values here are being collected and passed to ClowderConfigSource as existingValues, wouldn't the withoutExpansion have an impact for such existing value that should have been resolved. Within ClowderConfigSource it would be getting raw, non-resolved, value, wouldn't it?

This is correct and that's exactly what we need. Values should stay raw while being processed by clowder-quarkus-config-source. They will be resolved later though when your application will inject the values.

Reply:
Here's an example of existing value from one of our projects:
```
rbac/mp-rest/url=${clowder.endpoints.rbac-service:https://ci.cloud.redhat.com}
```
The `clowder.endpoints.rbac-service` expression refers to a configuration key produced by this project from the Clowder configuration. The `https://ci.cloud.redhat.com` part is a default value in case `clowder.endpoints.rbac-service` is not found.

Before this PR, `${clowder.endpoints.rbac-service:https://ci.cloud.redhat.com}` was immediately resolved before the `clowder.endpoints.rbac-service` value could even be produced. This means the `https://ci.cloud.redhat.com` was always returned. I had to work around that issue with [ClowderConfigInitializer](https://github.com/RedHatInsights/policies-ui-backend/blob/aeae3592dd8b278b2538a45ffded8abd6f8c0879/src/main/java/com/redhat/cloud/policies/app/ClowderConfigInitializer.java#L39-L43) in our app.

Now, `${clowder.endpoints.rbac-service:https://ci.cloud.redhat.com}` is passed raw to our application and when we inject the `rbac/mp-rest/url` key, the app resolves the expression, finds the `clowder.endpoints.rbac-service` value and correctly injects the URL from the Clowder config.